playbooks/roles/mongo_newrelic_monitor/tasks/main.yml

```diff
@@ -0,0 +1,24 @@
+---
+
+# TODO:
+# - create mongo listCollections role as per https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/mongodb-monitoring-integration#users-privileges
```

Comment:
I don't know if this is possible through ansible - is there a neat method to run an arbitrary mongo query?

Reply:
@swalladge I'm not sure about *neat* 😄 For creating this role we might need to fall back on sending a `mongo` CLI command via a `shell:` task.

For creating users there's [`mongodb_user`](https://docs.ansible.com/ansible/latest/modules/mongodb_user_module.html#mongodb-user-module) (community-maintained).

Reply:
Ok, I've worked on these parts. I have no idea what I'm doing lol.

Reply:
@swalladge Thanks, the most recent changes look good to me :)

For the remaining question about the step that creates the `listCollections` role, I wasn't sure what you meant by "doesn't respect MONGO_REPL_SET"?

Reply:
This just calls the mongo cli directly, which assumes localhost. The other mongo tasks take the `MONGO_REPL_SET` variable. I don't know if it's required or not, hence the comment. Is it ok to just use mongo cli directly?

Reply:
@swalladge Ah, I see.

> Is it ok to just use mongo cli directly?

I think so. In the absence of other options we also don't have much of a choice? ;)

> The other mongo tasks take the MONGO_REPL_SET variable. I don't know if it's required or not, ...

The command for creating the `listCollections` role doesn't reference the replica set so I'm assuming it's not required.

But we do want to be able to collect replica set statistics, so some extra steps might be necessary (cf. [MongoDB users and privileges](https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/mongodb-monitoring-integration#users-privileges)):

> Complete these steps on the mongos to be monitored. If mongod-level metrics are to be collected, such as host or replica set statistics, create the role and user on each mongod as well.

> This just calls the mongo cli directly, which assumes localhost.

If necessary, it looks like the replica set can be specified via the [`--hosts` parameter](https://docs.mongodb.com/manual/reference/program/mongo/#cmdoption-mongo-host) for the `mongo` command.

Reply:
True. This will work for us anyway.